README.md

```diff
@@ -14,54 +24,28 @@ New Relic requires the following environment variables to identify Kubernetes ob
-New Relic provides an easy method for deploying this automatic approach.
-
-## Setup
-
-### 1) Check if MutatingAdmissionWebhook is enabled on your cluster
-
-This feature requires Kubernetes 1.9 or later. Verify that the kube-apiserver process has the admission-control flag set.
-
-```
-$ kubectl api-versions | grep admissionregistration.k8s.io/v1beta1
-admissionregistration.k8s.io/v1beta1
-```
-
-### 2) Install the injection
+These environment variables are automatically injected in the pods using a MutatingAdmissionWebhook.
 
-```bash
-$ kubectl apply -f deploy/newrelic-metadata-injection.yaml
-```
+### Prerequisites
 
-Executing this:
+For the development process [Minikube](https://kubernetes.io/docs/getting-started-guides/minikube) and [Skaffold](https://github.com/GoogleCloudPlatform/skaffold) tools are used.
 
-- creates `newrelic-metadata-injection-deployment` and `newrelic-metadata-injection-svc`.
-- registers the `newrelic-metadata-injection-svc` service as a MutatingAdmissionWebhook with the Kubernetes API.
+* [Install Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/).
+* [Install Skaffold](https://github.com/GoogleCloudPlatform/skaffold#installation).
 
-### 3) Install the certificates
+Currently the project compiles with **Go 1.11.4**.
 
-This webhook needs to be authenticated by the Kubernetes extension API server, so it will need to have a signed certificate from a CA trusted by the extension API server. The certificate management is isolated from the webhook server and a secret is used to mount them. 
+### Dependency management
 
-**Important**: the webhook server has a file watcher pointed at the secret's folder that will trigger a certificate reload whenever anything is created or modified inside the secret. This allows easy certificate rotation with an update of the TLS secret that is created by running:
+[Go modules](https://github.com/golang/go/wiki/Modules) are used for managing dependencies. This project does not need to be in your GOROOT, if you wish so.
 
-```bash
-$ namespace=default # Change the namespace here if you also changed it in the yaml files.
-$ serverCert=$(kubectl get csr newrelic-metadata-injection-svc.${namespace} -o jsonpath='{.status.certificate}')
-$ tmpdir=$(mktemp -d)
-$ echo ${serverCert} | openssl base64 -d -A -out ${tmpdir}/server-cert.pem
-$ kubectl patch secret newrelic-metadata-injection-secret --type='json' \
-    -p "[{'op': 'replace', 'path':'/data/tls.crt', 'value':'$(serverCert)'}]"
-$ rm -rf $(tmpdir)
-```
+Currently for K8s libraries it uses version 1.13.1. Only couple of libraries are direct dependencies, the rest are indirect. You need to point all of them to the same K8s version to make sure that everything works as expected. For the moment this process is manual.
 
-#### Automatic management
+### Setup
 
-The certificate management can be automatic, using the Kubernetes extension API server (recommended, but optional):
+### Automatic certificate management
```

Comment:
Related to https://github.com/newrelic/k8s-metadata-injection/pull/46/files#r255859714.

I would include this section in the main "Setup" section.